internal/manager/config_validation_test.go

```diff
@@ -0,0 +1,77 @@
+package manager
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/controllers/gateway"
+)
+
+func TestConfigValidatedVars(t *testing.T) {
```

Comment:
The structure of these table (or maybe matrix) driven tests got my head spinning 🤯